File: LeetCode/413-arithmetic-slices/ex_413.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& A) {
        if (A.empty())  return 0;
        int diff = INT_MAX;
        int count = 0;
        int n = 0;
        for (int i = 1; i != A.size(); ++i) {
            if (diff != A[i] - A[i-1]) {
                diff = A[i] - A[i-1];
                count += (1 + n) * n / 2;
                n = 0;
            }
            else
                ++n;
        }
        if (n > 0)
            count += (1 + n) * n / 2;
        return count;
    }
};

File: LeetCode/413-arithmetic-slices/ex_413.py
class Solution(object):
    def numberOfArithmeticSlices(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        diff = sys.maxsize
        count = 0
        n = 0
        for i in xrange(1, len(A)):
            if diff != A[i] - A[i-1]:
                diff = A[i] - A[i-1]
                count += sum(xrange(n + 1))
                n = 0
            else:
                n += 1
        if n > 0:
            count += sum(xrange(n + 1))
        return count